src/ocaml-server/main.ml

```diff
@@ -122,32 +130,50 @@ let parse_folder_to_file =
 let rec listen msgBox=
   Lwt_mvar.take msgBox >>= fun events->
   Array.iter (fun event->
-    check_if_specific_file_mod (Event.t_to_string event))
+      check_if_specific_file_mod (Event.t_to_string event))
     events;
   flush stdout;
   listen msgBox
 
 let source_updates ()=
   match init_library () with
   | Status.FSW_OK->
-    let handle, msgBox= Fswatch_lwt.init_session Monitor.System_default in
-    add_path handle (String.concat "" ["./"; parse_folder_to_file]);
-    async (Fswatch_lwt.start_monitor handle);
-    listen msgBox
+     let handle, msgBox= Fswatch_lwt.init_session Monitor.System_default in
+     add_path handle (String.concat "" ["./"; parse_folder_to_file]);
+     async (Fswatch_lwt.start_monitor handle);
+     listen msgBox
   | err-> Lwt_io.eprintf "%s\n" (Status.t_to_string err)
 
-let data_updates () = 
-  match init_library () with
-  | Status.FSW_OK->
-    let handle, msgBox= Fswatch_lwt.init_session Monitor.System_default in
-    add_path handle "./../webpage/js/.";
-    async (Fswatch_lwt.start_monitor handle);
-    listen msgBox
-  | err-> Lwt_io.eprintf "%s\n" (Status.t_to_string err)
 
+exception PortInUseException of string
 
 let () =
-  print_endline "Server running, listening on port 3030 for HTTP requests\n http://127.0.0.1:3030\n";
-  let _ = Sys.command (String.concat "" ["mi "; Sys.argv.(1);" > ../webpage/js/data-source.json "]) in
-  Lwt_main.run (start_server () <&> source_updates () <&> data_updates ())
+  let speclist = [
+      "--no-file", Arg.Set(no_file), "Enables the server to run without a file";
+      "-p", Arg.Set_int(port), "Enable port choices.";
+    ] in
+  let anon_fun arg = visualize := arg in
+  let usage_msg = "Usage: either --no-file for client sending POST data or give full path to a source file. \n Note: one of them is mandatory!" in
+  Arg.parse speclist anon_fun usage_msg;
+  if !visualize = "" && not !no_file then
+    Arg.usage speclist usage_msg
+  else
+    begin
+      if not !no_file then
+        begin
+
+          ignore @@ Sys.command (String.concat "" ["mi "; !visualize ;" > ../webpage/js/data-source.json "]);
+          Lwt_main.run (Lwt.pick [source_updates ();
+                                  Lwt.catch
+                                    (fun () -> start_server !port ())
+                                    (function
+                                     | _ -> Lwt.fail (PortInUseException (String.concat "" ["Error: Port "; (string_of_int !port); " already in use. Run the same command again with a different port. " ])))])
```

Comment:
It seems like you are handling all exceptions by raising a self defined `PortInUseException`. But what if `start_server` happens to throw a different error? Why not match against the exception `Unix.Unix_error(Unix.EADDRINUSE, _, _)` and handling it separately, while just pass along any other exception?

Reply:
Fixed in latest commit